compiler/rustc_expand/src/mbe/macro_check.rs

```diff
@@ -269,12 +272,20 @@ fn check_binders(
                 // FIXME: Report this as a hard error eventually and remove equivalent errors from
                 // `parse_tt_inner` and `nameize`. Until then the error may be reported twice, once
                 // as a hard error and then once as a buffered lint.
-                psess.buffer_lint(
-                    MISSING_FRAGMENT_SPECIFIER,
-                    span,
-                    node_id,
-                    BuiltinLintDiag::MissingFragmentSpecifier,
-                );
+                if psess.edition >= Edition::Edition2024 {
```

Comment:
This breaks edition hygiene and uses global `psess.edition` instead of `span.edition()`, so 2024 edition code won't be able to use old crates with missing metavar kinds.
But that should be fine if it's eventually going to turn into a hard error anyway.

Reply:
Changed to `span.edition()` to be safe.